### DISCUSSION TITLE
What's the best practice for loading multi collections into memory

### DISCUSSION THREAD

**Discussion by iFurySt** [milvus-io/milvus#33251] [Q&A and General discussion] [answered]
Hi, we have multi-collections, we need to query all around the collection according to the caller, but it happens at random times. So we load all collections now. Is it the best practice for loading collections into memory?

**Comment by yhmo** [comment, marked as answer]
Loading process is time-consuming, keeping all collections in memory is for the best search performance.
If you want to swap collections between storage and memory, you can create an upper layer on SDK to manage all the search requests, and apply LRU or LFU logic to swap collections.

**Reply by iFurySt** [reply]
Good advice! thanks a lot.
